packages/compass-telemetry/src/generic-track.ts

```diff
@@ -0,0 +1,68 @@
+import { type Logger, mongoLogId } from '@mongodb-js/compass-logging';
+
+export type TrackProps = Record<string, any> | (() => Record<string, any>);
+export type TrackFunction = (event: string, properties?: TrackProps) => void;
+
+export interface TelemetryPreferences {
+  getPreferences(): { trackUsageStatistics: boolean };
+}
+
+export interface TelemetryServiceOptions {
+  sendTrack: (event: string, properties: TrackProps) => void;
+  logger?: Logger;
+  preferences?: TelemetryPreferences;
+}
+
+export const createTrack = ({
+  sendTrack,
+  logger: { log, debug },
+  preferences,
+}: TelemetryServiceOptions & { logger: Logger }) => {
+  const trackAsync = async (
+    event: string,
+    properties: TrackProps = {}
+  ): Promise<void> => {
+    // Note that this preferences check is mainly a performance optimization,
+    // since the main process telemetry code also checks this preference value,
+    // so it is safe to fall back to 'true'.
+    const { trackUsageStatistics = true } = preferences?.getPreferences() ?? {};
+
+    if (!trackUsageStatistics) {
+      return;
+    }
+
+    if (typeof properties === 'function') {
+      try {
+        properties = await properties();
+      } catch (error) {
+        // When an error arises during the fetching of properties,
+        // for instance if we can't fetch host information,
+        // we track a new error indicating the failure.
+        // This is so that we are aware of which events might be misreported.
+        sendTrack('Error Fetching Attributes', {
+          event_name: event,
+        });
+        log.error(
+          mongoLogId(1_001_000_313),
```

Comment:
We usually try to preserve the log ids between events, this one was different before

```suggestion
          mongoLogId(1_001_000_190),
```

Reply:
I don't know why, but eslint in vscode is telling me it's duplicated. but running eslint on the package works, so I can keep that.